spec/multitenant_spec.rb

```diff
@@ -43,6 +43,22 @@ class Item < ActiveRecord::Base
   describe 'Multitenant.current_tenant' do
     before { Multitenant.current_tenant = :foo }
     it { Multitenant.current_tenant == :foo }
+    it 'accesses current_tenant in thread safe manner' do
+      in_tenants = [:foo, :bar, :tar]
+      out_tenants = []
+      threads = []
+
+      count = 2
+      count.times do |i|
+        threads << Thread.new(i) do |index|
+          Multitenant.current_tenant = in_tenants[i+1]
+          out_tenants[i+1] = Multitenant.current_tenant
+        end
+      end
+      threads.each(&:join)
+      out_tenants[0] = Multitenant.current_tenant
+      out_tenants.should == in_tenants
```

Comment:
Is this comparison order independent? I assume that since the threads are tiny they execute in order but it's not guaranteed, right?

Reply:
order dependent - but that is what we want.

Reply:
Right, the array is accessed by index so the order of values in it is independent of the thread run order.